simulation/tests/test_model.py

```diff
@@ -54,6 +54,32 @@ def test_increment_timestep_updates_boiler_upgrade_scheme_cumulative_spend_gbp(
         model.increment_timestep()
         assert model.boiler_upgrade_scheme_cumulative_spend_gbp == 22_000
 
+    def test_air_source_heat_pump_discount_factor_is_zero_if_no_discount_schedule_passed(
+        self,
+    ):
+
+        model = model_factory(air_source_heat_pump_price_discount_schedule=[])
+
+        assert model.air_source_heat_pump_discount_factor == 0
+
+    def test_air_source_heat_pump_discount_factor_changes_when_crosses_discount_schedule_date(
+        self,
+    ):
+
+        model = model_factory(
+            start_datetime=datetime.datetime(2022, 2, 1),
+            air_source_heat_pump_price_discount_schedule=[
+                (datetime.datetime(2022, 2, 1), 0),
```

Comment:
Just because we have already tested for 0 in a previous test so we want to make sure its working correctly
```suggestion
                (datetime.datetime(2022, 2, 1), 0.1),
```